### DISCUSSION TITLE
Moq Nagware -- What is the Board's Response

### DISCUSSION THREAD

**Discussion by gortok** [dotnet-foundation/Home#87] [Board of Directors]
I [wrote about the Moq Nagware issue](https://georgestocker.com/2023/08/09/moq-adds-nagware-wheres-the-net-foundation-in-all-this/), and I have two questions for the Board:

1. What is your response to this situation?
2. Given the mission of the .NET foundation, you all sit in the hot-seat for open source sustainability in .NET.  What is your plan to address the open source sustainability issue that caused the Moq maintainer to add this nagware?

**Comment by glennawatson** [comment]
It's a item on the projects committee meeting for tomorrow so feel free to join if you can. Would love the feedback. 

I don't think we have a official official position yet but everyone is pretty concerned. I think one of my goals is to develop a analytics best practice guide that we can use to judge new projects joining the foundation by and as a set of recommendations for existing projects. 

As moq isn't a foundation project we would hope the guide would help enforce better practices by them. 

Given moq has kind of forced this on us I guess it'll be a priority in our meeting today as well. 

I'll get you details and links to that meeting if you're interested. It's at 21:00 utc. 

Edit: doh I got the date wrong. Time zone and being well ahead of the US.

**Comment by glennawatson** [comment]
In terms of your post some inaccuracies. 

> Also, all but one committee stopped meeting (wg-projects, wg-maintainers, wg-marketing, wg-membership, wg-corporate-relations, wg-education have all stopped meeting/posting their minutes/updating agendas) — the wg-outreach team is the only committee to have met recently.

Committees are meeting regularly. The thing is I went hunting for meeting details. if I wasn't getting invites by the foundation I wouldn't know the times and where and how to join.  Something I need to fix. Also we hire a company at the moment to take notes and minutes and I need to discover where that is going. I will get back to you on this one too.

**Reply by glennawatson** [reply]
Ok. 

So some of the committees you need to apply to join. To the projects one is one of them. The projects one has information in the readme updated as of yesterday. 

https://github.com/dotnet-foundation/projects

Joining the Committee

The Project Committee meets for one hour on the 2nd Thursday of every month at 5:00 PM ET via Microsoft Teams. If you are interested in becoming a member of our team, please submit a request to join the project team and express your interest in being added to the meeting invite. We welcome individuals from all backgrounds and experience levels who share our passion for .NET, the open source model, and collaborative software development.

**Reply by glennawatson** [reply]
Also https://dotnetfoundation.org/about/committees

Phew. The maintainers committee was combined into the projects about two months ago.

**Comment by gortok** [comment]
@glennawatson I appreciate your responses; I'll update the post with your edits.

Taking your responses at face value, the next concern that brings up is; these committees are active, but as far as I can tell, there's been no output (deliverables, plans, ideas) out of these committees, which raises the issue... what have they been _doing_?  (clearly, minutes and agendas are one way to learn that information, another is the results of the work the committee does). If it's meet, take minutes, but no actions are taken, then if I were a member of that committee, I'd be rather annoyed and probably vote with my feet.

>Given moq has kind of forced this on us I guess it'll be a priority in our meeting today as well.
> I'll get you details and links to that meeting if you're interested. It's at 21:00 utc.

I appreciate that invite; it's rather short notice, but where is the meeting location advertised? If I wanted to drop in as a dues paying member of the foundation, can I?

I noticed that the .NET Foundation Newsletter repo hasn't been updated in a few years, I would have expected information about committee meetings to be posted there; it's on the website, but that lists the meeting dates as the "Second Thursday of the month" for the projects committee.  Forgive me, but yesterday was not the second Thursday of the month.

From an organizational perspective; most DNF repos have not been updated since October 2021, and that's a concern from a "is this thing even viable" standpoint.  For the sake of understanding, I'm happy to take your words at face value, but for an online organization, if there's no updates, is work even happening?

This is the second time I've asked the .NET Foundation to make its position known on this sort of issue, the last was with "Hot Reload", and the answer there was the same, "We can't have an opinion on a member/non-member organization doing something that has a negative effect on the .NET community"

Frankly, that viewpoint dooms the Foundation to... managing meetups?  [This is a .NET Interest group, and as I've explained before](https://georgestocker.com/2021/10/30/the-fundamental-problem-the-net-foundation-board-has/) at least by the nature of your by-laws and mission, and I expect you to take an interest in .NET issues. I expect you to lead. I expect you to further .NET.  You can't do that if you don't have opinions, don't have a vision, and don't have leadership in the .NET space.

**Reply by glennawatson** [reply]
> I appreciate that invite; it's rather short notice, but where is the meeting location advertised? If I wanted to drop in as a dues paying member of the foundation, can I?

https://github.com/dotnet-foundation/projects#joining-the-committee is for the projects committee. Current conducted as a teams meeting for those who apply and then put on a email chain. I screwed up saying yesterday btw since being in Australia I'm way ahead timezone wise of the US and it is today. I joined a empty meeting by mistake yesterday. 

Other committees. https://dotnetfoundation.org/about/committees

In terms of what's been happening with them. The foundation has been paying for secretarial series (a large amount but we given notice we are planning on ending those services) which put meeting notes behind a SharePoint internal website. I'm aiming to get those public since like you said they should be showing output. 

Board members will volunteer their time now is the plan to make meeting minutes. We should be publishing on GitHub for the community. 

I came in as a board member this month and I want to produce tangible results for the maintainer community that is financially stable. That's realistically going to limit what can be done. 

Any other questions let me know. I can't promise particular results yet since I'm getting my feet.

**Reply by gortok** [reply]
> Current conducted as a teams meeting for those who apply and then put on a email chain. I screwed up saying yesterday btw since being in Australia I'm way ahead timezone wise of the US and it is today. I joined a empty meeting by mistake yesterday.

Why are the meetings so... secretive?

Joining is about the only open part of the process.

Once you're in, the location of the meeting is secret, the minutes are secret, the actions of the committee are secret, and even the recordings (if they exist) of the meeting themselves are... secret -- that is, not open to public inspection (or even 'semi-public' inspection by dues-paying members of the .NET Foundation.

From what you've said in this thread, this doesn't seem to be intended. if all of this is because you're using what technologies are provided by your largest member organization, then it would appear those technologies are not purpose-built for a group that otherwise hosts all of its innards on github; the face of public work.

Heck, as far as I can tell, the Slack is also not used very much at all -- only one channel had recent activity (newer than 90 days, thanks Salesforce).

You've got far more important things to do than to worry about what one person sees; but visibly, after the hot reload and Copyright assignment debacle, the foundation's public work all but... died. 

[No work to fix the by-laws](https://github.com/dotnet-foundation/Home/discussions/60), [no work to give the dues paying members more understanding of Microsoft's role in the governance of the organization](https://github.com/dotnet-foundation/Home/discussions/63), nothing. 

You're a new board member, and I know from experience what it's like to pick up as an executive in a non-profit organization where terms end and there's very little infrastructural support for succession, so I empathize with your situation, but take this as what it is: someone who cares enough to bring up the issue and show a perspective on what is perhaps the biggest opportunity for .NET to move forward as a community and platform.

**Reply by jguadagno** [reply]
> Why are the meetings so... secretive?

The meetings are not secretive, in the past we just did not have a way, or the idea of just making them open and in public.

The Foundation has spent the last 12 months or so cleaning up our "back office" stuff, vendors, contracts, where things live, etc. while trying to still provide support for the ever-changing landscape.

In the coming weeks/months, we are going to be working on cleaning up our communication, which encompasses meeting invites/agendas/minutes,  our git repos, and our website.

**Reply by jguadagno** [reply]
As far as Slack, Discord, Teams, add your favorite social platform, it's been a challenge to keep up with all of them, some like Slack, some like Discord, some like git,  it's hard to be everywhere, and that lies the first problem.  We tried to be everywhere and have done a less than ideal job of keeping them all up to date or usable.  This is something we plan to work on
